### PR TITLE
Fix all unknown block not outputted 

### LINKED ISSUE #1805: Plan output would miss a dynamic block if it is fully unknown
### OpenTofu Version

```shell
1.8.0-beta
```

### OpenTofu Configuration Files

```hcl
resource "random_integer" "a" {
  max = 10
  min = 1
}

resource "azurerm_firewall_policy" "random_policy" {
  name                = "random-policy"
  resource_group_name = "existing-resource-group"
  location            = "westeurope"

  dynamic "dns" {
    for_each = random_integer.a.result > 5 ? [] : [1]
    content {
      servers = ["8.8.8.8"]
    }
  }
}
```

### Debug Output

.

### Expected Behavior

OpenTofu should show the `dns` block as `(known after apply)`

### Actual Behavior

OpenTofu never shows the `dns` block as part of the creation, whether or not it is created

```
OpenTofu used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

OpenTofu will perform the following actions:

  # azurerm_firewall_policy.random_policy will be created
  + resource "azurerm_firewall_policy" "random_policy" {
      + child_policies           = (known after apply)
      + firewalls                = (known after apply)
      + id                       = (known after apply)
      + location                 = "westeurope"
      + name                     = "random_policy"
      + resource_group_name      = "existing-resource-group"
      + rule_collection_groups   = (known after apply)
      + sku                      = (known after apply)
      + threat_intelligence_mode = "Alert"
    }

  # random_integer.a will be created
  + resource "random_integer" "a" {
      + id     = (known after apply)
      + max    = 10
      + min    = 1
      + result = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

```

### Steps to Reproduce

1. `tofu init`
2. `tofu plan`

### Additional Context

This is reproducible with any resource that has a block field

### References

- Related issue in the Terraform Repository - https://github.com/hashicorp/terraform/issues/34933

### PR DESCRIPTION
**Pull request by buraksenn**
Resolves #1805 

Plan diff needs to show nested unknown blocks but it does not. Thus, this PR implements required change for it

## Target Release

1.9.0

## Checklist
- [X] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [X] I have not used an AI coding assistant to create this PR.
- [X] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.
- [X] I (and other contributors to this PR) have not looked at the Terraform source code while implementing this PR.

### Go checklist
- [X] I have run golangci-lint on my change and receive no errors relevant to my code.
- [X] I have run existing tests to ensure my code doesn't break anything.
- [X] I have added tests for all relevant use cases of my code, and those tests are passing.
- [X] I have only exported functions, variables and structs that should be used from other packages.
- [X] I have added meaningful comments to all exported functions, variables, and structs.

### CODE REVIEW

File: internal/command/jsonformat/differ/block.go [hunk 1]
```diff
@@ -70,45 +70,50 @@ func ComputeDiffForBlock(change structured.Change, block *jsonprovider.Block) co
...
-		afterSensitive := childValue.IsAfterSensitive()
-		forcesReplacement := childValue.ReplacePaths.Matches()
+		beforeSensitive := childChange.IsBeforeSensitive()
+		afterSensitive := childChange.IsAfterSensitive()
+		forcesReplacement := childChange.ReplacePaths.Matches()
+
+		if diff, ok := checkForUnknownBlock(childChange, block); ok {
+			blocks.AddSingleBlock(key, diff, forcesReplacement, beforeSensitive, afterSensitive)
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (ollevche) [reply internal/command/jsonformat/differ/block.go:87]: I think we also need to check if the diff is zero to skip it. The same way as we do [here](https://github.com/opentofu/opentofu/pull/1948/files#diff-84d5de04e60ac3b7d0cdb2597b40d9f6dee730bddb8d0073ea330e510a1a4981R115) for example.

**Review by ollevche — COMMENTED** [review]
Thank you for the PR! Overall looks nice, left a minor enhancement comment.

File: internal/command/jsonformat/differ/block.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (buraksenn) [reply internal/command/jsonformat/differ/block.go]: Added the check before adding unknown block. Thanks for the review

**Review by ollevche — APPROVED** [review]

**Review by cam72cam — APPROVED** [review]

**Comment by cam72cam** [comment]
Thanks @buraksenn, sorry for the delay

**Pull request merged**